config/setup.go

```diff
@@ -0,0 +1,54 @@
+package config
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+	pb "go.viam.com/api/service/slam/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// SetupDirectories creates the data directory at the specified path along with
+// its data, map, and config subdirectories.
+func SetupDirectories(dataDirectory string, logger golog.Logger) error {
+	for _, directoryName := range [4]string{"", "data", "map", "config"} {
+		directoryPath := filepath.Join(dataDirectory, directoryName)
+		if _, err := os.Stat(directoryPath); err != nil {
+			logger.Warnf("%v directory does not exist", directoryPath)
```

Comment:
Can we please log the error & change the log to not say the directory doesn't exist if it might fail for a different reason.

Reply:
How do you feel about the following logs: 

```
    logger.go:130: 2023-03-07T23:44:22.970Z	WARN	config/setup.go:24	setup directories: stat /tmp/2898068953/data: no such file or directory
    logger.go:130: 2023-03-07T23:44:22.970Z	WARN	config/setup.go:24	setup directories: stat /tmp/2898068953/map: no such file or directory
    logger.go:130: 2023-03-07T23:44:22.971Z	WARN	config/setup.go:24	setup directories: stat /tmp/2898068953/config: no such file or directory
```


And for invalid permissions:
```
    logger.go:130: 2023-03-07T23:44:22.971Z	WARN	config/setup.go:24	setup directories: stat /tmp/3520687813/no_permissions/data: permission denied
```

Reply:
that seems good. Is that output from an unpushed change? I ask b/c I don't see any log on line 24 in the current code in this branch.

Reply:
It is on line 27 now. Same warning format.